src/components/Widgets/NumberControl.js

```diff
@@ -1,17 +1,34 @@
 import React, { PropTypes } from 'react';
 
-export default class StringControl extends React.Component {
+export default class NumberControl extends React.Component {
   handleChange = (e) => {
-    this.props.onChange(e.target.value);
+    const valueType = this.props.field.get('valueType')
+    let value
+    if(valueType === 'int') {
+      value = parseInt(e.target.value, 10)
+    } else if(valueType === 'float') {
+      value = parseFloat(e.target.value)
+    } else {
+      value = e.target.value
+    }
+    this.props.onChange(value);
   };
 
   render() {
-    return <input type="number" id={this.props.forID} value={this.props.value || ''} onChange={this.handleChange} />;
+    const { field, value, forID } = this.props
+    const min = field.get('min')
+    const max = field.get('max')
+    const step = field.get('step')
+    return <input type="number" id={forID} value={value || ''} step={step == null ? (field.get('valueType') === 'int' ? 1 : '') : step} min={min == null ? '' : min} max={max == null ? '' : max} onChange={this.handleChange} />;
```

Comment:
Also, you can simply pass a second argument to `.get` in order to set a default value if the value you're looking for isn't present, which should allow you to remove most of the `||` and ternaries which are currently checking for defaults.